### PR TITLE
Some ideas around making dev setup easier 

### PR DESCRIPTION
**Pull request by alxndrsn**
1. use node-foreman to build vue / watch for changes / start nginx
2. make `nginx.conf` less dependent on os-specific details / require less manual configuration
3. update README setup instructions to reflect 1 & 2

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -13,13 +13,104 @@
...
+      video/mpeg                            mpeg mpg;
+      video/quicktime                       mov;
+      video/webm                            webm;
+      video/x-flv                           flv;
+      video/x-m4v                           m4v;
+      video/x-mng                           mng;
+      video/x-ms-asf                        asx asf;
+      video/x-ms-wmv                        wmv;
+      video/x-msvideo                       avi;
+  }
```
Comment (alxndrsn) [inline nginx.conf:113]: I doubt this is all necessary - it's just a copy of my local `mime.types`

File: nginx.conf [hunk 2]
```diff
@@ -62,9 +153,7 @@ http {
     }
 
     location / {
-      # Specify the dist/ directory of the repository to `nginx -p` so that . is
-      # relative to dist/.
-      root .;
+      root ../dist;
```
Comment (alxndrsn) [inline nginx.conf:156]: Now running from `tmp/`.  Could also run from the project root, and then explicitly put logs in `tmp/`.

File: package-lock.json [hunk 3]
```diff
@@ -757,70 +757,6 @@
...
-          "resolved": "https://registry.npmjs.org/loader-utils/-/loader-utils-2.0.2.tgz",
-          "integrity": "sha512-TM57VeHptv569d/GKh6TAYdzKblwDNiumOdkFnejjD0XwTH87K90w3O7AiJRqdQoXygvi1VQTJTLGhJl7WqA7A==",
-          "dev": true,
-          "optional": true,
-          "requires": {
-            "big.js": "^5.2.2",
-            "emojis-list": "^3.0.0",
-            "json5": "^2.1.2"
-          }
-        },
```
Comment (alxndrsn) [inline package-lock.json:823]: :shrug:

File: nginx.conf [hunk 4]
```diff
@@ -13,13 +13,104 @@
 #
 # One difference between this file and the configuration for production is that
 # this configuration does not support request body decompression.
+daemon off;
+worker_processes 1;
+error_log stderr;
+pid ./tmp/nginx.pid;
```
Comment (matthew-white) [inline nginx.conf:19]: Are these changes needed? I think it's really helpful to minimize the number of differences between this file and the nginx configuration in [prod](https://github.com/getodk/central/blob/master/files/nginx/odk.conf.template).

File: nginx.conf [hunk 5]
```diff
@@ -13,13 +13,104 @@
...
 
 events {
 }
 http {
-  # This path may be different on your local computer depending on your OS and
-  # how you installed nginx.
-  include /usr/local/etc/nginx/mime.types;
+  access_log ./tmp/nginx-access.log;
```
Comment (matthew-white) [inline nginx.conf:24]: Similar as above, is this addition needed?

File: nginx.conf (hunk 1 above)
Comment (matthew-white) [reply nginx.conf:113]: Sounds good. :+1: I like the idea of specifying these inline rather than including a file whose path changes based on the OS.

File: nginx.conf (hunk 2 above)
Comment (matthew-white) [reply nginx.conf:156]: Is `tmp/` just for storing logs? For better or for worse, I'm actually not sure that I've ever used the logs from this local nginx.

File: package-lock.json (hunk 3 above)
Comment (matthew-white) [reply package-lock.json:823]: Sorry, I'm hoping that soon we won't have to deal with this! These unrelated changes should go away if you remove `node_modules`, then run `npm install` again.

File: README.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ First, install Node.js 14+.
 
 Next, install dependencies by running `npm install`.
 
-Install NGINX. Depending on your operating system and how you install NGINX, you may need to change the absolute paths in the development [`nginx.conf`](/nginx.conf).
+Install NGINX. Depending on your operating system and how you install NGINX.
```
Comment (matthew-white) [inline README.md:36]: Maybe we can now remove the second sentence entirely?

**Review by matthew-white — COMMENTED** [review]
Thank you, @alxndrsn! I'm definitely excited about changes that could make Frontend development easier.

> 1. use node-foreman to build vue / watch for changes / start nginx
> 2. make `nginx.conf` less dependent on os-specific details / require less manual configuration

(2) feels like a big win to me!

The value of (1) feels slightly less clear. Is the benefit there that the user doesn't need to start nginx manually (thereby running one command instead of two)? I think I could get behind that, especially if it's not a huge code change. I think it would be nice to minimize the changes to nginx.conf.

**Comment by matthew-white** [comment]
Oh, and it'd be awesome if commit messages were a little more descriptive! Especially the commit message of the first commit, since if the PR is squash-merged, I think GitHub will use that message in the commit message it suggests for the squash-merged commit. You could also consider doing some of that squashing / commit cleanup locally, before pushing. Or if you've already pushed, you could do that locally, then force-push.

File: README.md (hunk 6 above)
Comment (alxndrsn) [reply README.md:36]: :grin:  yes, it makes no sense now!

File: nginx.conf (hunk 2 above)
Comment (alxndrsn) [reply nginx.conf:156]: It stores the access log, and the nginx pid file.  I think ideally the pidfile wouldn't be necessary, and the logs would go to stdout...

Comment (alxndrsn) [reply nginx.conf:156]: Ref pidfile, there isn't much in the nginx docs: https://nginx.org/en/docs/ngx_core_module.html#pid

This thread looks relevant, but doesn't suggest there's a reliable & convenient alternative: https://github.com/docker-library/httpd/issues/166

File: nginx.conf (hunk 4 above)
Comment (alxndrsn) [reply nginx.conf:19]: `daemon off`: this means that nginx will run as a foreground process, which seems a lot less surprising for frontend devs not familiar with nginx

`worker_processes 1`: this is the default, so can be removed

`error_log stderr`: i think this is both necessary and helpful - it removes one point of OS-specificity, and means that devs will see nginx errors in their terminal rather than have to tail the error log file

`pid ./tmp/nginx.pid`: again this removes another place where nginx may default to an absolute path which the current user does not have access to

File: nginx.conf (hunk 5 above)
Comment (alxndrsn) [reply nginx.conf:24]: Yes, similarly to `error_log`, this may otherwise default to an absolute path which a normal user may not have permissions for, and if they are using nginx for other purposes may pollute their logs.

**Comment by alxndrsn** [comment]
@matthew-white I've separated the nginx config changes into their own PR at https://github.com/getodk/central-frontend/pull/621.

If that looks like it'll be merged, I can update this PR to be foreman-specific.

**Comment by alxndrsn** [comment]
Ideas addressed in separate PRs.
